### ISSUE TITLE
Recipe and Summoning Have Never Been Fixed

### ISSUE THREAD

**Issue by Boolyman** [Brian-Wuest/MC-FromTheDepths#6]
The original issue never resolved.  The FTD_summons folder still shows up empty on Linux servers.  I constantly have to zip and send files over to people to replace the empty folder.  There is no FTD_Summons folder appearing in the instance folder anymore, that was fixed... but the one inside the config is still installing empty.

**Comment by Brian-Wuest** [comment]
so the .json files are not showing up then?

I downloaded your server files and they do appear to be in there. Do they report this after starting up the server or as soon as they copy the files over?

Are there any log files for me to dig through? I have added a bunch of logging in the last few updates.

From your server files, it doesn't appear as though you are using the latest version. I added a fix for when the FTD_Summons directory doesn't exist and removed a bunch of old code and added more logging.

I actually added a new file (1.1.0.4) which addresses an issue where the files are not read if the file encoding doesn't match the default one on the machine. They need to be UTF-8 encoded now. That's probably not your issue though.

**Comment by Boolyman** [comment]
Gonna have them update, and see if it still happens, then get me some logs.

**Comment by Boolyman** [comment]
Seems the issue is still recurring.

I saw this in the log file, does it mean anything?  
[06:38:36] [main/ERROR]: From_The_Depths: Parsing error loading recipe from_the_depths:spawninfo. com.google.gson.JsonSyntaxException: Expected a com.google.gson.JsonObject but was com.google.gson.JsonArray

**Comment by Brian-Wuest** [comment]
That's interesting. It is trying to load the spawninfo file as a recipe. It might be doing that incorrectly if the case doesn't match what I have in code. When I get home I can check quick. That would explain why none of the recipes load.

**Comment by Brian-Wuest** [comment]
Yeap, it's an issue with character casing. I was looking for "spawnInfo" you had "spawninfo" so it didn't match what I was looking for. I will be changing the code to be case insensitive. In the mean time you can change it to "spawnInfo".

**Comment by Brian-Wuest** [comment]
Version 1.1.0.5 has been uploaded which addresses this issue.

**Comment by Boolyman** [comment]
I have no idea what you are talking about lol.  What do you mean I have "spawninfo" change it to "spawninfo"... change what?  The log?  The folder? The file?

**Comment by Brian-Wuest** [comment]
change the file name from "spawninfo" (note the lower-case i) to "spawnInfo" (the upper-case i). Or just include the latest update which has been approved on curseforge :).

**Comment by Boolyman** [comment]
I get 2-3 of these a day.  Im lost for what can be done.

![2018-05-25 04_35_00-greenshot](https://user-images.githubusercontent.com/24765883/40542460-1f3e2180-5fd5-11e8-908e-5ee4f4a400f1.png)

**Comment by Brian-Wuest** [comment]
Do you have log files from the people reporting issues?

**Comment by Brian-Wuest** [comment]
Something I noticed when I downloaded your config files. All of the recipes use the "minecraft:barrier" for their ingredients. I am guessing you have a mod which does some funky magic to the recipe so they have to do something in order to unlock the recipe.

I am not sure if that works (just in general) since I never tested with that. When they create the item in this way does it still have the correct name? (Aphrodite, Zeus, Biff, etc) in the UI? I use NBT tags for determining which monster to summon and if the recipe is modified incorrectly it would be the cause for this not working.

For the Altar, I also have a check in there for not allowing the summoning if the world's difficulty is set to Peaceful. I highly doubt it is, but just throwing that out there.

Either way, I need log files in order to try to figure this out.

**Comment by Boolyman** [comment]
No there is no mod that interacts with recipes or totems.  They are simply given to the players in a quest.  Not craftable.  Which is why their recipe contains an item only available in creative.  The Biff Totem IS craftable and also can't be summoned on server.

**Comment by Boolyman** [comment]
I will, as usual try to get some logs.  But it's difficult when I dont have a server, and most of the people that report the issue arent server admins themselves.  At this point I may just forget about it and leave it broken... it's already been a critical issue throughout the main pack release, the main youtubers that were doing playthroughs have probably/hopefully found a way to gloss over the issue.

**Comment by Brian-Wuest** [comment]
In a quest you say? What mod is doing that and how is it setup?

**Comment by Boolyman** [comment]
Its Better Questing Mod, It has tasks and rewards... I select the reward to give them from JEI.  Once the player completes the task, they get that reward automatically.  The capsules have the correct names on them, and people have even tried to use totems they pull directly from JEI since they didnt know if the issue was just with the totem they got from the quest.

**Comment by Brian-Wuest** [comment]
ah, darn :(

**Comment by Boolyman** [comment]
This is the message I got from the latest server having the problem:

I guess the Log will not help at all, i found no information in it... how ever...
https://paste.ubuntu.com/p/WmWFCtpsVn/
Its the complete log from starting client, joining single player world, "Open to LAN" giving both items and trying to summon.
The funny part is, its working on single player as long as i dont start it as a server by "Open to LAN". On the hostet server it doesnt work either

If there are special parameters to get a more detailed log, please tell me what to do

OK i think i got a better log now, i tried to switch the log4j2.xml from my forge***.jar from info to debug.
I am not very used to java itself, more c++ so not sure if it helped but here is another, pretty large log
https://paste.ubuntu.com/p/gGxMcYPHdQ/

I think the interesting part ist here:
[18:53:05] [Server thread/TRACE] [FML]: Sending event FMLInitializationEvent to mod from_the_depths
[18:53:05] [Server thread/WARN] [from_the_depths]: From_The_Depths: Summoning recipe found at location [/mnt/config/FTD_Summons/summon_Hades.json] has output which doesn't contain valid nbt data or specifies a boss which doesn't exist.
[18:53:05] [Server thread/WARN] [from_the_depths]: From_The_Depths: Summoning recipe found at location [/mnt/config/FTD_Summons/summon_Biff.json] has output which doesn't contain valid nbt data or specifies a boss which doesn't exist.
[cut out]
[18:53:05] [Server thread/TRACE] [FML]: Sent event FMLInitializationEvent to mod from_the_depths(edited)

**Comment by Brian-Wuest** [comment]
That's very interesting. I can try to duplicate that on my windows machine. May not get to that until much later tonight or tomorrow with it being fathers day.

**Comment by Brian-Wuest** [comment]
Welp here is another attempt to fix this issue. I discovered that the way I was attempting to resolve the "spawnInfo" file was not case insensitive and the bosses were not loading properly which means then the items were not loading properly (on one side or the other).

I have also added some better logging for when the recipes are being loaded for the bosses to allow me to determine which part of the recipe output is bad. Such as is there a mis-spelled boss name or if the nbt data is malformed. I am also outputting the names of each boss loaded from the spawnInfo.json file to help narrow down issues where a certain boss totem isn't loading.

Thank you for providing that log as it has helped me narrow down the issue.

EDIT: Version 1.1.0.6 is available on CurseForge.
